hy/core/macros.hy

```diff
@@ -507,6 +507,42 @@
   `(if-not ~test (do ~@body)))
 
 
+(defmacro do-n [count-form &rest body]
+  "Execute `body` a number of times equal to `count-form` and return
+  ``None``. (To collect return values, use :hy:macro:`list-n`
+  instead.) Negative values of the count are treated as 0.
+
+  This macro is implemented as a :hy:macro:`for` loop, so you can use
+  :hy:macro:`break` and :hy:macro:`continue` in the body.
+
+  ::
+
+     => (do-n 3 (print \"hi\"))
+     hi
+     hi
+     hi
+  "
+  `(for [~(gensym) (range ~count-form)]
+    ~@body))
+
+
+(defmacro list-n [count-form &rest body]
+  "Like :hy:macro:`do-n`, but the results are collected into a list.
+
+  ::
+
+    => (setv n 0)
+    => (list-n 5 (+= n 1) n)
+    [1 2 3 4 5]
```

Comment:
Maybe change the variable `n` to something else in this example?

It seems that `list-n` suggests that `n` is somehow special, which could confuse someone reading this at a glance. If it was changed to `i` or something I think that it would be slightly clearer.

Reply:
Fair enough; changed. I also factored out some code into a subroutine.